src/borg/archiver.py

```diff
@@ -258,11 +258,18 @@ def do_init(self, args, repository):
                 'for details about the security implications.', shlex.quote(path))
 
         if key.NAME != 'plaintext':
-            logger.warning(
-                '\n'
-                'IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!\n'
-                'Use "borg key export" to export the key, optionally in printable format.\n'
-                'Write down the passphrase. Store both at safe place(s).\n')
+            if 'repokey' in key.NAME:
+                logger.warning(
+                    '\n'
+                    'IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!\n'
+                    'The key is already included in the repository config.\n'
```

Comment:
as long as nothing goes wrong in the repo config, the key is there.

but it would be somehow ridiculous, if the repo config gets corrupted (hw or sw error), and you lose access to all these years of backups you've made, just because you did not make a backup of the key.

so, i do not see it as an improvement NOT to recommend "borg key export" here.

Reply:
Makes sense.

Just a thought: That's something that could be improved on the side of borg (for example, by storing the key multiple times in the repo).